karapace/avro_compatibility.py

```diff
@@ -52,19 +51,23 @@ class SchemaCompatibilityResult:
     def __init__(
         self,
         compatibility: SchemaCompatibilityType = SchemaCompatibilityType.recursion_in_progress,
-        incompatibilities: List[SchemaIncompatibilityType] = None,
+        incompatibilities: Optional[List[SchemaIncompatibilityType]] = None,
```

Comment:
Note that the annotation for `Optional` is not required when used with a default value `= None`, ref:

https://mypy.readthedocs.io/en/stable/command_line.html#none-and-optional-handling

I usually prefer to be explicit, this way I can apply just one heuristic on annotations. This also has the side benefit of allowing me to easily copy the type to datastructures (e.g. if I'm refactoring a class to a dataclass, or creating a namedtuple to call a function in a threadpool)